tests/test_jmcomic/test_jm_feature.py

```
@@ -0,0 +1,142 @@
+        # PluginFeature 根据来源推导执行时机
+        pf = Feature.export_pdf
+        # download_album → 只在 after_album 执行
+        self.assertTrue(pf.should_invoke('download_album', 'after_album'))
+        self.assertFalse(pf.should_invoke('download_album', 'after_photo'))
+        # download_photo → 只在 after_photo 执行
+        self.assertTrue(pf.should_invoke('download_photo', 'after_photo'))
+        self.assertFalse(pf.should_invoke('download_photo', 'after_album'))
+
+    def test_adapt_kwargs(self):
+        """测试 PluginFeature 参数动态适配"""
+        # download_album 模式：P前缀 → A前缀, level → album
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `：` (FULLWIDTH COLON). Did you mean `:` (COLON)?

(RUF003)